test/tests/pbs_accumulate_resc_used.py

```diff
@@ -0,0 +1,1456 @@
+             'reserve_start': time.time() + 10,
+             'reserve_end': time.time() + 30, }
+        r = Reservation(TEST_USER, a)
+        rid = self.server.submit(r)
+        a = {'reserve_state': (MATCH_RE, "RESV_CONFIRMED|2")}
+        self.server.expect(RESV, a, id=rid)
+
+        rname = rid.split('.')
+        # Submit a job inside reservation
+        a = {'Resource_List.select': '3:ncpus=1', ATTR_queue: rname[0]}
+        j = Job(TEST_USER)
+        j.set_attributes(a)
+        jid = self.server.submit(j)
+
+        # Verify the resource values
+        a = {'resources_used.foo_i': '6',
+             'resources_used.foo_i2': '3kb',
+             'resources_used.foo_f': '3.06',
+             'resources_used.foo_f2': '6.03',
+             'resources_used.stra': "\"happy\"",
+             'resources_used.stra2': "\"glad\"",
+             'job_state': 'F'}
+        self.server.expect(JOB, a, extend='x',
+                           offset=30, interval=1,
+                           max_attempts=20, id=jid)
+
+        # Restart server and verifies that the values are still the same
+        self.server.restart()
+
+        # Below is commented due to a known PBS issue
+        # self.server.expect(JOB, a, extend='x', id=jid)
+
+    def test_server_restart(self):
+        """
+        Test that resource accumulation will not get
+        impacted if sever is restarted during job execution
+        """
+
+        # Create a prologue hook
+        hook_body = """
+import pbs
+e=pbs.event()
+pbs.logmsg(pbs.LOG_DEBUG, "executed prologue hook")
+if e.job.in_ms_mom():
+    e.job.resources_used["vmem"] = pbs.size("11gb")
+    e.job.resources_used["foo_i"] = 11
+    e.job.resources_used["foo_f"] = 0.11
+    e.job.resources_used["foo_str"] = '{"seven":7}'
+    e.job.resources_used["cput"] = 11
+    e.job.resources_used["stra"] = '"glad,elated","happy"'
+    e.job.resources_used["foo_str4"] = "eight"
+else:
+    e.job.resources_used["vmem"] = pbs.size("12gb")
+    e.job.resources_used["foo_i"] = 12
+    e.job.resources_used["foo_f"] = 0.12
+    e.job.resources_used["foo_str"] = '{"eight":8,"nine":9}'
+    e.job.resources_used["cput"] = 12
+    e.job.resources_used["stra"] = '"cucumbers,bananas"'
+"""
+
+        hook_name = "prolo"
+        a = {'event': "execjob_prologue", 'enabled': 'True'}
+        self.server.create_import_hook(
+            hook_name,
+            a,
+            hook_body,
+            overwrite=True)
+
+        a = {'Resource_List.select': '3:ncpus=1',
+             'Resource_List.walltime': 10,
+             'Resource_List.place': 'scatter'}
+        j = Job(TEST_USER)
+        j.set_attributes(a)
+        j.set_sleep_time("5")
+
+        # The pbsdsh call is what allows a first task to get spawned on
+        # on a sister mom, causing the execjob_prologue hook to execute.
+        j.create_script(
+            "pbsdsh -n 1 hostname\n" +
+            "pbsdsh -n 2 hostname\n" +
+            "sleep 300\n")
```

Comment:
Yes, the j.set_sleep_time() is unnecessary. I'll update the walltime request to 300.